### PR TITLE
CUDNN grouped convolutions + depthwise convolution

### PR DESCRIPTION
**Pull request by wendazhou**
This pull request implements grouped convolutions backed by the [CUDNN 7](http://docs.nvidia.com/deeplearning/sdk/cudnn-release-notes/Chunk1484074616.html#rel_7) convolution groups feature.

The feature is exposed in the DNN support class and the Conv2d ops launchers, but no API / operations are created to instantiate grouped convolutions directly.

This pull request also implements dispatching the DepthwiseNativeConv2d (and the corresponding backpropagation operations) to these new grouped convolution kernels. This feature is gated behind a feature flag TF_DEPTHWISE_CONV_USE_GROUPED_CONV which is disabled by default. This increases performance slightly for training Mobilenet in single precision (about 5% on Volta), but also paves the way for float16 training (which is very slow with the current implementation).

Todo: (need advice / directions):
- add tests for the new DepthwiseNativeConv2d path (probably duplicate existing tests but will also need to set the environment variable - how to do this?)
- the feature flag may not be completely satisfactory at the moment: users compiling on CUDNN 6 and setting the feature flag will face a somewhat cryptic error that "no algorithms are found" as the shapes will be incorrect for the conv2d kernels.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @benoitsteiner: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by chsigg** [comment]
Hi Wenda. Thank you very much for your contribution, and sorry for the delay. We are actively working on mapping depthwise convolution to cuDNN's grouped convolution, and it should become available in the coming days. Unfortunately, it has been taking longer than expected. Instead of using a feature flag, we implemented this as a separate, 'labeled' kernel.

I will look into your change and see how it compares to our implementation. Thank you again!

**Comment by wendazhou** [comment]
Thanks for your work! I am looking forward to see an official implementation of Depthwise and Grouped convolutions in tensorflow.

**Comment by chsigg** [comment]
cuDNN's grouped convolutions to perform depthwise convolution can now be enabled with [graph._kernel_label_map](https://github.com/tensorflow/tensorflow/blob/8cb0558da924e891aa1bb5d79a6c0c846301e4eb/tensorflow/python/framework/ops.py#L3311)({"DepthwiseConv2dNative": "cudnn_grouped_convolution"}).

In the future, we will automatically choose between TF's depthwise convolution and cuDNN's grouped convolution, whichever gives the better performance.

**Comment by austingg** [comment]
@chsigg  is there some speed comparasion about DepthwiseConv2Native when used cudnn, how much improved?
